proof_of_concept/definitions/workflows.py

```diff
@@ -19,12 +20,23 @@ def __init__(
             inputs: Dict mapping input parameter names to references to
                     their sources, either the name of a workflow input,
                     or of the form other_step.output_name.
-            outputs: List of names of outputs produced.
+            outputs: Dict mapping output parameter names to references
+                    to the base assets to use, or None if containers
+                    are not used.
             compute_asset_id: The id of the compute asset to use.
         """
         self.name = name
         self.inputs = inputs
-        self.outputs = outputs
+        self.outputs = dict()   # type: Dict[str, Optional[Identifier]]
+
+        for name, base_asset in outputs.items():
+            if base_asset is None:
+                self.outputs[name] = None
+            elif not isinstance(base_asset, Identifier):
+                self.outputs[name] = Identifier(base_asset)
+            else:
+                self.outputs[name] = base_asset
```

Comment:
Yes, those code paths are independent. ComputeMetaData is made when a compute asset is created, a WorkflowStep is made when a workflow is created. Those are done by different people at different times, in general. Eventually we may want to make it so that the WorkflowStep is filled in automatically from the metadata, but at the moment the steps are made directly, in particular in the test cases where having to write `Identifier(...)` all the time is annoying.